### PR TITLE
Add support for STM32F429ZI chip and NUCLEO-429ZI board

### PR DESCRIPTION
**Pull request by garasubo**
### Pull Request Overview

This pull request is adding support for NUCLEO-429ZI board (#1263 is a previous version).
Tock has support for NUCLEO-446RE board (#1220 ), which is very similar to NUCLEO-429ZI. So, I refactor the original support packages to make it easy to apply for other STM boards.

### Testing Strategy
I used libtock-c repository to test my change in NUCLEO-429ZI. But I haven't test it on NUCLEO-446RE because I don't have that board. 

-     examples/button
-     examples/c_hello
-     examples/tests/console_timeout

### TODO or Help Wanted
I haven't test on NUCLEO-446RE board, so please help me test my change.
This is my first PR, so please let me know if I miss something.

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make formatall`.

### CODE REVIEW

File: boards/nucleo_f429zi/Makefile [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Makefile for building the tock kernel for the NUCLEO-446RE platform
```
Comment (bradjc) [inline boards/nucleo_f429zi/Makefile]: Update

File: boards/nucleo_f429zi/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,358 @@
+//! Board file for Nucleo-F446RE development board
+//!
+//! - <https://www.st.com/en/evaluation-tools/nucleo-f446re.html>
```
Comment (bradjc) [inline boards/nucleo_f429zi/src/main.rs]: update

File: boards/nucleo_f429zi/src/virtual_uart_rx_test.rs [hunk 3]
```diff
@@ -0,0 +1,87 @@
+//! Test reception on the virtualized UART by creating two readers that
```
Comment (bradjc) [inline boards/nucleo_f429zi/src/virtual_uart_rx_test.rs]: I imagine we don't need a copy of this?

**Comment by rajivr** [comment]
> @rajivr are you willing/able to help review and test this?

Yes, I'll take a look at this PR shortly. Sorry for not being able to get to it sooner.

File: chips/stm32f446re/src/usart.rs [hunk 4]
```diff
@@ -477,36 +498,33 @@ impl hil::uart::Uart<'a> for Usart<'a> {}
...
+                // alert client
+                self.rx_client.map(|client| {
+                    buffer.map(|buf| {
+                        client.received_buffer(
+                            buf,
+                            length,
+                            ReturnCode::SUCCESS,
+                            hil::uart::Error::None,
+                        );
                     });
```
Comment (rajivr) [inline chips/stm32f446re/src/usart.rs]: @garasubo Thank you for implementing this. After thinking bit more about the macro issue, I wanted to implement exactly the same thing, so we could avoid introducing the macro. :-) I had planned on refactoring `transfer_done` and test it out the change week before suggesting potential change to #1263, but looks like you got to it sooner. Nice!

**Comment by rajivr** [comment]
@alevy I've reviewed and tested this series. Good to go from my side! Thank you @garasubo for the awesome work!

File: boards/nucleo_f429zi/Makefile (hunk 1 above)
Comment (garasubo) [reply boards/nucleo_f429zi/Makefile]: Done

File: boards/nucleo_f429zi/src/main.rs (hunk 2 above)
Comment (garasubo) [reply boards/nucleo_f429zi/src/main.rs]: Done

File: boards/nucleo_f429zi/src/virtual_uart_rx_test.rs (hunk 3 above)
Comment (garasubo) [reply boards/nucleo_f429zi/src/virtual_uart_rx_test.rs]: Oh, yes. This is not architecture specific. Removed.

**Comment by garasubo** [comment]
Hi, do I need to do anything else to merge this request?

**Comment by alevy** [comment]
@garasubo, no, just someone needs to pull the trigger, so here we go:

bors r+

**Comment by garasubo** [comment]
Cool. Thank you so much for all your help! I'm very excited that now I can use Tock on my board!

**Pull request merged**